src/main/features/response/routes/your-payment-plan.ts

```diff
@@ -0,0 +1,46 @@
+import * as express from 'express'
+
+import { Paths } from 'response/paths'
+
+import { ErrorHandling } from 'common/errorHandling'
+import { Form } from 'app/forms/form'
+import { DraftService } from 'services/draftService'
+import { User } from 'idam/user'
+import { PaidAmount } from 'ccj/form/models/paidAmount'
+import { DefendantPaymentPlan } from 'response/form/models/defendantPaymentPlan'
+import { FormValidator } from 'forms/validation/formValidator'
+
+function renderView (form: Form<PaidAmount>, res: express.Response): void {
+  const user: User = res.locals.user
+  const alreadyPaid: number = user.responseDraft.document.paidAmount.amount || 0
+  res.render(Paths.defenceFullPartialPaymentPlanPage.associatedView, {
+    form: form,
+    remainingAmount: user.claim.totalAmount - alreadyPaid
+  })
+}
+
+/* tslint:disable:no-default-export */
+export default express.Router()
+  .get(Paths.defenceFullPartialPaymentPlanPage.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+      const user: User = res.locals.user
+      renderView(new Form(user.responseDraft.document.defendantPaymentPlan), res)
+    }))
+
+  .post(Paths.defenceFullPartialPaymentPlanPage.uri,
+    FormValidator.requestHandler(DefendantPaymentPlan, DefendantPaymentPlan.fromObject),
+    ErrorHandling.apply(
+      async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+
+        const form: Form<DefendantPaymentPlan> = req.body
+        const user: User = res.locals.user
+
+        if (form.hasErrors()) {
+          renderView(form, res)
+        } else {
+          const { externalId } = req.params
+          user.responseDraft.document.defendantPaymentPlan = form.model
+          await new DraftService().save(user.responseDraft, user.bearerToken)
+          res.redirect(Paths.checkAndSendPage.evaluateUri({ externalId: externalId }))
```

Comment:
Should not we take users to task list again so that they can go to check and send manually. This is the first time I see this behaviour. What is the point for taking user to check and send in case they did not complete the whole task list?

Reply:
Ok I agree